### PR TITLE
[doc] Remove use_frameworks! usage for RN 0.62

### PR DESCRIPTION
**Pull request by jfroffice**
the use_frameworks! does not work anymore the workaround

### CODE REVIEW

**Comment by mfazekas** [comment]
@jfroffice thanks for the PR.

Can you clarify 
```
the use_frameworks! does not work anymore
```
?!

`use_frameworks! ` is still recommended by mapbox, so i think we should keep it the recommended solution and documenting the way to disable it.

**Comment by jfroffice** [comment]
I didn't succeed to get it works on with `xcode v11.4`.

The `pod install` failed
```
[!] The 'Pods-XXXXXX' target has transitive dependencies that include statically linked binaries: 
.../ios/Pods/CocoaLibEvent/lib/libevent.a
.../ios/Pods/CocoaLibEvent/lib/libevent_core.a, 
.../ios/Pods/CocoaLibEvent/lib/libevent_extra.a,
.../ios/Pods/CocoaLibEvent/lib/libevent_pthreads.a,
.../ios/Pods/OpenSSL-Universal/ios/lib/libcrypto.a, 
.../ios/Pods/OpenSSL-Universal/ios/lib/libssl.a
```

**Comment by jfroffice** [comment]
I'd test on `RN 0.61.0` and it's working like a charm.

The issue is on `RN 0.62.0`.

**Comment by mfazekas** [comment]
@jfroffice you tested with a en empty rn app, or were there other deps too?!

**Comment by FrederickEngelhardt** [comment]
Encountering this error as well. I am able to use flipper still with the !use_frameworks flag with the  `react-native-permissions` workaround

Solution:

1. Add the code below to your pod file
1. Go into your project and add a `dummy.swift` file. You can just leave a comment to remove later.
1. Run `pod install`
1. CMD+SHIFT+K (to clean) and rebuild the app

In your podfile add the following.
```
def add_flipper_pods!
  version = '~> 0.33.1'
  pod 'FlipperKit', version, :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitLayoutPlugin', version, :configuration => 'Debug'
  pod 'FlipperKit/SKIOSNetworkPlugin', version, :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitUserDefaultsPlugin', version, :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitReactPlugin', version, :configuration => 'Debug'
end
# Post Install processing for Flipper
def flipper_post_install(installer)
  installer.pods_project.targets.each do |target|
    if target.name == 'YogaKit'
      target.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '4.1'
      end
    end
  end
  file_name = Dir.glob("*.xcodeproj")[0]
  app_project = Xcodeproj::Project.open(file_name)
  app_project.native_targets.each do |target|
    target.build_configurations.each do |config|
      cflags = config.build_settings['OTHER_CFLAGS'] || '$(inherited) '
      unless cflags.include? '-DFB_SONARKIT_ENABLED=1'
        puts 'Adding -DFB_SONARKIT_ENABLED=1 in OTHER_CFLAGS...'
        cflags << '-DFB_SONARKIT_ENABLED=1'
      end
      config.build_settings['OTHER_CFLAGS'] = cflags
    end
    app_project.save
  end
  installer.pods_project.save
end

# Workaround for use_frameworks with this module https://github.com/react-native-community/react-native-permissions
use_frameworks!

$dynamic_frameworks = ['Mapbox-iOS-SDK', 'MapboxMobileEvents', 'React', 'React-Core', 'Folly', 'boost-for-react-native', 'DoubleConversion', 'glog', 'React-cxxreact', 'React-jsinspector','React-jsi', 'React-jsiexecutor', 'Yoga', 'React-RCTActionSheet', 'React-RCTAnimation', 'React-RCTBlob', 'React-RCTNetwork', 'React-RCTImage', 'React-RCTLinking','React-RCTSettings', 'React-RCTText', 'React-RCTVibration', 'RCTTypeSafety', 'FBLazyVector', 'RCTRequired', 'ReactCommon', 'FBReactNativeSpec']

pre_install do |installer|
  installer.pod_targets.each do |pod|
    if !$dynamic_frameworks.include?(pod.name)
      puts "Overriding the static_framework? method for #{pod.name}"
      def pod.build_type;
        Pod::Target::BuildType.static_library
      end
    end
  end
end
# end of workaround for https://github.com/react-native-community/react-native-permissions
...
target '[your_app_name]Tests' do
    inherit! :complete # possibly broken and needs to be swapped back to :search_paths
    # Pods for testing
  end
```

**Comment by FrederickEngelhardt** [comment]
@jfroffice let me know if the above solution helps.

**Comment by jfroffice** [comment]
I update the MR.

@FrederickEngelhardt  are you using the `RN 0.62.0`

I'd noticed that the current `./exampls/ios/Podfile` is already using the fallback.
https://github.com/react-native-mapbox-gl/maps/blob/master/example/ios/Podfile

File: ios/install.md [hunk 1]
```diff
@@ -1,20 +1,27 @@
 # iOS Installation
 
-## Using CocoaPods
+## React-Native >= `0.62.0`
```
Comment (mfazekas) [inline ios/install.md:3]: The only difference in 0.62 and 0.61 is that for 0.62 `use_frameworks!` does not work by default. 
We'd need to document that they should either add `use_frameworks!` or the workaround. We can also mention that the default 0.62 project is not `use_frameworks!` compatible if flipper is enabled and so they should be using the workaround in that case.

**Comment by mfazekas** [comment]
@jfroffice  see also #784. That would revers the default to non `use_frameworks!` that would allow it to work with autolink on ios for RN 0.61, 0.62. There is still option to use `use_frameworks!` variant.

**Comment by mfazekas** [comment]
@jfroffice thanks for looking into it. I've now changed the default setting to work with RN 0.62. So no `use_frameworks!` is required by default. I've also updated the README.md please check and submit a new PR if needed.

**Comment by jfroffice** [comment]
@mfazekas, OK great !

**Comment by FrederickEngelhardt** [comment]
> I update the MR.
> 
> @FrederickEngelhardt are you using the `RN 0.62.0`
> 
> I'd noticed that the current `./exampls/ios/Podfile` is already using the fallback.
> https://github.com/react-native-mapbox-gl/maps/blob/master/example/ios/Podfile

Yes, more specifically `0.62.1`

**Pull request closed without merging**